pallets/runtime/src/asset.rs

```diff
@@ -435,26 +441,16 @@ decl_module! {
 
             ensure!(total_supply <= MAX_SUPPLY.into(), Error::<T>::TotalSupplyAboveLimit);
 
-            // Alternative way to take a fee - fee is proportionaly paid to the validators and dust is burned
-            let validators = <pallet_session::Module<T>>::validators();
-            let fee = Self::asset_creation_fee();
-            let validator_len:T::Balance;
-            if validators.is_empty() {
-                validator_len = T::Balance::from(1 as u32);
-            } else {
-                validator_len = T::Balance::from(validators.len() as u32);
-            }
-            let proportional_fee = fee / validator_len;
-            // for v in validators {
-            //     <balances::Module<T> as Currency<_>>::transfer(
-            //         &sender,
-            //         &<T as Utils>::validator_id_to_account_id(v),
-            //         proportional_fee,
-            //         ExistenceRequirement::AllowDeath
-            //     )?;
-            // }
-            let remainder_fee = fee - (proportional_fee * validator_len);
-            let _withdraw_result = <balances::Module<T>>::withdraw(&sender, remainder_fee, WithdrawReason::Fee.into(), ExistenceRequirement::KeepAlive)?;
+            // The fee is proportionaly paid to the validators and dust is burned.
+            let fee_recipients: Vec<T::AccountId> = <pallet_session::Module<T>>::validators()
+                .into_iter()
+                .map(T::ValidatorIdToAccountId::convert)
+                .collect();
+            <<T as IdentityTrait>::ProtocolFee>::charge_fee_equal_parts(
+                &signer,
+                &OperationName::from(protocol_op::ASSET_CREATE_TOKEN),
+                fee_recipients.as_slice()
+            )?;
```

Comment:
We don't need a special case here. The normal charge fee function should be called. The earlier code was a very hacky poc code from quite a while back.